wallet/src/program_facades/amm.rs

```diff
@@ -0,0 +1,451 @@
+use common::{error::ExecutionFailureKind, rpc_primitives::requests::SendTxResponse};
+use nssa::{AccountId, program::Program};
+
+use crate::{
+    WalletCore,
+    cli::account::TokenHolding,
```

Comment:
Hm, I think something's wrong here... Ideally I'd like to have `cli` as a thing used only in `main.rs` stuff. Not `lib.rs` stuff.

Maybe someday even make `cli` module to be included only with `cli` feature, although there are still unresolved UX [issues](https://github.com/rust-lang/cargo/issues/4663) in rustlang.